management/src/test/java/org/ehcache/docs/ManagementTest.java

```diff
@@ -78,26 +78,27 @@ public void usingManagementRegistry() throws Exception {
       aCache.get(0L);
       aCache.get(0L);
 
-      Thread.sleep(1000);
-
       Context context = StatsUtil.createContext(managementRegistry); // <5>
 
-      ResultSet<ContextualStatistics> counters = managementRegistry.withCapability("StatisticsCapability") // <6>
+      StatisticQuery query = managementRegistry.withCapability("StatisticsCapability") // <6>
           .queryStatistic("Cache:HitCount")
           .on(context)
-          .build()
-          .execute();
+          .build();
+
+      long onHeapHitCount;
+      do {
```

Comment:
Could we add a comment explaining why we are looping ?
Something like : 
```
// it could be several seconds before the sampled stats could be available, let's try until we find the correct value : 4
```